lib/extensions/base.py

```diff
@@ -223,3 +227,45 @@ def get_base_file_name(self):
     def uniqueId(self, prefix, make_new_id=True):
         """Override inkex.Effect.uniqueId with a nicer naming scheme."""
         return generate_unique_id(self.document, prefix)
+
+    def _update_inkstitch_document(self, nodes):
+        # get the inkstitch svg version from the document
+        svg = self.document.getroot()
+        search_string = "//*[local-name()='inkstitch_svg_version']//text()"
+        file_version = svg.findone(search_string)
+        if file_version is None:
+            file_version = 0
+        else:
+            file_version = int(file_version)
+
+        if file_version == INKSTITCH_SVG_VERSION:
+            return
+
+        if file_version > INKSTITCH_SVG_VERSION:
+            inkex.errormsg(_("This document was created with a newer Version of Ink/Stitch. "
+                             "It is possible that not everything works as expected.\n\n"
+                             "Please update your Ink/Stitch version: https://inkstitch.org/docs/install/"))
+            # they may not want to be bothered with this info everytime they call an inkstitch extension
+            # let's udowngrade the file version number
+            self._update_inkstitch_svg_version()
+        else:
+            # this document is either a new document or it is outdated
+            # if we cannot find any inkstitch attribute in the document, we assume that this is a new document which doesn't need to be updated
+            search_string = "//*[namespace-uri()='http://inkstitch.org/namespace' or " \
+                            "@*[namespace-uri()='http://inkstitch.org/namespace'] or " \
+                            "@*[starts-with(name(), 'embroider_')]]"
+            inkstitch_element = svg.findone(search_string)
+            if inkstitch_element is None:
```

Comment:
I think we want to update if there _are_ existing Ink/Stitch attributes in the doc, right? So this should be `inkstitch_element is not None`?

Reply:
Yes, but we still need to set the inkstitch svg file version if we do not want to run the update check again.